File: Operation_Map.Server/Models/Subcontractor.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Text.RegularExpressions;


namespace Operation_Map.Server.Models
{
    public class Subcontractor
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? Id { get; set; }
        public string? ContactName { get; set; }
        public string? CompanyName { get; set; }
        public string? CompanyType { get; set; }
        public double? PercentagePaid { get; set; }
        public string? PdfUpload { get; set; }  // URL to the uploaded PDF
        public string? Address { get; set; }
        public List<SubProject>? Projects { get; set; }
        public string? ContactPhoneNumber { get; set; }
        public string? Website { get; set; }
        public string? ContactEmail { get; set; }
        public string? Ubi { get; set; }  // Unified Business Identifier
        public string? License { get; set; }
        public string? TaxId { get; set; }
        public string? Other {  get; set; }
        public string? InsuranceCompany { get; set; }
        public string? InsuranceAgent { get; set; }
        public string? InsurancePhone { get; set; }
        public string? InsuranceEmail { get; set; }
        public bool? AdditionalInsured { get; set; }
        public string? CertificatePDFURL { get; set; }
        public string? AdditionalInsurancePDFURL { get; set; }
    }

}

File: app/shared/data/subcontractors/subcontractor.ts
import { Contact } from './../project/project-list';

export interface Subcontractor {
  id: string;
  contactName?: string;
  companyName?: string;
  companyType?: string;
  percentagePaid?: number;
  pdfUpload?: string; // URL to the uploaded PDF
  address?: string;
  bids?: number;
  contactPhoneNumber?: string;
  website?: string;
  contactEmail?: string;
  subProjects?: any[];
  ubi?: string; // Unified Business Identifier
  license?: string;
  taxId?: string;
  other?: string;
  insuranceCompany?: string;
  insuranceAgent?: string;
  insurancePhone?: string;
  insuranceEmail?: string;
  additionalInsured?: boolean;
  certificatePDFURL?: string;
  additionalInsurancePDFURL?: string;
  dateStarted?: Date;
  dateFinished?: Date;
  paidInvoices?: string[];
  bidDate?: Date;
  bidPrice?: number;
  paid?: number;
  leftToPay?: number;
  bidPDFUrl?: string;
  materialCost?: number;
  totalCost?: number;
  selectedBid?: boolean;
}
